### PR TITLE
Put --use-cache on all actions

### PR DESCRIPTION
**Pull request by Oddant1**

### CODE REVIEW

**Comment by Oddant1** [comment]
~~Blocked by qiime2/qiime2#771~~

**Comment by Oddant1** [comment]
Some of the diff here is updating some other errors to be less ugly. Sorry that's kinda unrelated to the rest of the PR

**Comment by Oddant1** [comment]
Closes #333

**Review by colinvwood — APPROVED** [review]
- [x] write discussed test
- [x] continue editing @colinvwood's comments

@colinvwood y'know I can just outright edit this comment even though you left it. Kinda wack.

**Review by colinvwood — APPROVED** [review]

File: q2cli/core/artifact_cache_global.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ def set_used_artifact_cache(args):
 
 
 def unset_used_artifact_cache():
-    """Set the USED_ARTIFACT_CACHE back to the None.
+    """Set the USED_ARTIFACT_CACHE back to None.
```
Comment (colinvwood) [inline q2cli/core/artifact_cache_global.py:61]: I mean None is a singleton so `the None` works

**Review by colinvwood — APPROVED** [review]

File: q2cli/core/artifact_cache_global.py (hunk 1 above)
Comment (Oddant1) [reply q2cli/core/artifact_cache_global.py:61]: I did consider that while making the commit; however, I am not willing to go that hard

**Pull request merged**